src/main/java/nl/novi/techiteasy/config/SpringSecurityConfig.java

```diff
@@ -0,0 +1,96 @@
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.ProviderManager;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+/*  Deze security is niet de enige manier om het te doen.
+    In de andere branch van deze github repo staat een ander voorbeeld
+ */
+@Configuration
+@EnableWebSecurity
+public class SpringSecurityConfig {
+
+    public final CustomUserDetailsService customUserDetailsService;
+
+    private final JwtRequestFilter jwtRequestFilter;
+
+    public SpringSecurityConfig(CustomUserDetailsService customUserDetailsService, JwtRequestFilter jwtRequestFilter) {
+        this.customUserDetailsService = customUserDetailsService;
+        this.jwtRequestFilter = jwtRequestFilter;
+    }
+
+    // PasswordEncoderBean. Deze kun je overal in je applicatie injecteren waar nodig.
+    // Je kunt dit ook in een aparte configuratie klasse zetten.
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+
+    // Authenticatie met customUserDetailsService en passwordEncoder
+    @Bean
+    public AuthenticationManager authenticationManager(HttpSecurity http, PasswordEncoder passwordEncoder) throws Exception {
+        var auth = new DaoAuthenticationProvider();
+        auth.setPasswordEncoder(passwordEncoder);
+        auth.setUserDetailsService(customUserDetailsService);
+        return new ProviderManager(auth);
+    }
+
+
+
+
+    // Authorizatie met jwt
+    @Bean
+    protected SecurityFilterChain filter (HttpSecurity http) throws Exception {
+
+        http
+                .csrf(csrf -> csrf.disable())
+                .httpBasic(basic -> basic.disable())
+                .cors(Customizer.withDefaults())
+                .authorizeHttpRequests(auth ->
+                                auth
+                                        // Publieke endpoints
+                                        // Todo fix: getting the following error 500:"An unexpected error occurred: Handler dispatch failed: java.lang.NoSuchMethodError: 'void org.springframework.web.method.ControllerAdviceBean.<init>(java.lang.Object)'"
+                                        .requestMatchers("/api-docs/**", "/swagger-ui/**", "/v3/api-docs/**").permitAll()
```

Comment:
getting the following error 500:"An unexpected error occurred: Handler dispatch failed: java.lang.NoSuchMethodError: 'void org.springframework.web.method.ControllerAdviceBean.<init>(java.lang.Object)'"

<img width="1574" alt="Screenshot 2025-01-19 at 20 56 30" src="https://github.com/user-attachments/assets/f9afce86-3c36-4d7c-a77c-09334c22395c" />


Heb je een idee hoe kan ik deze error overkomen?

Reply:
Het werk wil in mijn ander project.